glfw/build.zig

```diff
@@ -36,14 +36,25 @@ pub const Options = struct {
     linux_window_manager: LinuxWindowManager = .X11,
 };
 
-pub fn link(b: *Builder, step: *std.build.LibExeObjStep, options: Options) void {
+/// Extend a library by statically linking glfw with it
+pub fn linkStaticLibrary(b: *Builder, other_lib: *std.build.LibExeObjStep, options: Options) void {
```

Comment:
Hi there @Avokadoen ! Really cool stuff here, thanks a ton for the PR - I'm quite excited to try out zig_vulkan now :)

I'd like to understand more what the changes to this build.zig do exactly so I can learn from you - my expectation was that anyone outside of the `mach` repo could already do this:

1. Import `const glfw = @import("glfw/build.zig")`
2. `glfw.link(b, myLibOrExe, .{})`

Based on these changes, though, it seems like either (1) that did not work for you or (2) there was an issue with "calling `glfw.link(...)` leads to the entire library rebuilding"? Is my guess for either of those right?

As an aside, I'd like to keep the external name just `glfw.link` instead of `glfw.linkStaticLibrary` for a few reasons:

1. I think static linking should be the default.
2. I think if dynamic linking is ever an option (I am not particularly interested in it), that it could be an option under the `Options` struct.

What do you think?

Reply:
Hi!
Let me just start by thanking you for the amazing work on the glfw building script and wrapper API.
It seems like calling ``step.linkLibrary(lib)`` when ``lib`` has been initialized through ``var lib = b.addStaticLibrary("foo", "baz")`` makes the final build command do ``zig build-lib`` even when the desired command is ``zig build-exe``

The naming of the functions could be improved as the linkStaticLibrary function is referencing linking the glfw step with a another static library step i.e mach engine (linking with mach is done statically if i'm note mistaken?).

I agree dynamic linking is not very useful in the context of this project!

Reply:
After trying to create a minimum requirement to reproduce the error it seems to work with a simple hello world linked using the existing link function. I'm converting the PR to a draft until I can submit an issue explaining things. I'll look into it tomorrow :^)

Reply:
This is more likely an issue debate rather than a PR debate at this point. It seems like a fresh zig project created with `ìnit-exe`` does not suffer form the side effect of being ran with ``build-lib`` like the zig_vulkan project. Currently unsure if this is a mach, zig_vulkan, or a zig compiler bug and so I'm hesitant to submit a issue on this project at this point. I will not prioritize this issue as my fork works for now. I'll submit an issue at a later point when I eventually get back to this.